doc/whatsnew/2.14.rst

```diff
@@ -281,10 +281,6 @@ Other Changes
 
   Closes #6644
 
-* ``pylint.pyreverse.ASTWalker`` has been removed, as it was only used internally by a single child class.
-
-  Ref #6712
-
```

Comment:
This is now under "Deprecations" in order to be grouped with #6713, although this is technically not a deprecation but a hard removal. Let me know if that's OK for you to put it there.